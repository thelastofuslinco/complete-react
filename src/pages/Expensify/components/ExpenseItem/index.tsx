import { Component } from 'react'
import ExpenseForm from '../ExpenseForm'
import { ExpenseModel } from '../../../../models/ExpenseModel'

interface State {
  isEdit: boolean
}

interface Props extends ExpenseModel {}

class ExpenseItem extends Component<Props, State> {
  state = {
    isEdit: false
  }

  render() {
    return this.state.isEdit ? (
      <li className="expenseContainer">
        <ExpenseForm
          {...this.props}
          onClick={() => this.setState({ isEdit: false })}
        />
      </li>
    ) : (
      <li className="expenseContainer">
        <span>Description: {this.props.description}</span>
        <span>Amount: {this.props.amount}</span>
        <span>Note: {this.props.note}</span>
        <span>
          CreatedAt: {new Date(this.props.createdAt).toLocaleString('en-us')}
        </span>
        <button onClick={() => this.setState({ isEdit: true })}>Edit</button>
      </li>
    )
  }
}

export default ExpenseItem
