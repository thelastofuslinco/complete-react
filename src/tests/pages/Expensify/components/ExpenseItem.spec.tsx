import { render, screen } from '@testing-library/react'
import { expenses } from '../../../fixtures/expenses'
import ExpenseItem from '../../../../pages/Expensify/components/ExpenseItem'

describe('Expense item component', () => {
  test('should render expense item', () => {
    render(<ExpenseItem {...expenses[0]} />)
    const element = screen.getAllByRole('listitem')
    expect(element).toMatchSnapshot()
  })
})
