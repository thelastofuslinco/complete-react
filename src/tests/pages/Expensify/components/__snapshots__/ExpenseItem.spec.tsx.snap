// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense item component should render expense item 1`] = `
[
  <li
    class="expenseContainer"
  >
    <span>
      Description: 
      any_description
    </span>
    <span>
      Amount: 
      822
    </span>
    <span>
      Note: 
      any_description
    </span>
    <span>
      CreatedAt: 
      11/5/2023, 12:00:00 AM
    </span>
    <button>
      Edit
    </button>
  </li>,
]
`;
